### PR TITLE
transfer (total escrow): add escrow / unescrow functions

### PR DESCRIPTION
**Pull request by crodriguezvega**
## Description

ref: #3368 

### Commit Message / Changelog Entry

```bash
N/A
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [x] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

**Review by colin-axner — APPROVED** [review]
LGTM, I added a little more code documentation

**Review by DimitrisJim — APPROVED** [review]
should these target the [feat/total-escrow-state-entry](https://github.com/cosmos/ibc-go/tree/feat/total-escrow-state-entry) branch?

**Comment by colin-axner** [comment]
> should these target the [feat/total-escrow-state-entry](https://github.com/cosmos/ibc-go/tree/feat/total-escrow-state-entry) branch?

No, since this is just code-hygenie/readability, I think it makes sense not to backport

File: modules/apps/transfer/keeper/relay.go [hunk 1]
```diff
@@ -397,6 +366,42 @@ func (k Keeper) refundPacketToken(ctx sdk.Context, packet channeltypes.Packet, d
 	return nil
 }
 
+// escrowToken will send the given token from the provided sender to the escrow address. It will also
+// update the total escrowed amount by adding the escrowed token to the current total escrow.
+func (k Keeper) escrowToken(ctx sdk.Context, sender, escrowAddress sdk.AccAddress, token sdk.Coin) error {
+	if err := k.bankKeeper.SendCoins(
+		ctx, sender, escrowAddress, sdk.NewCoins(token),
+	); err != nil {
```
[the remark this answers is not in the document]
Comment (colin-axner) [reply modules/apps/transfer/keeper/relay.go:374]: I thought about the same, agreed!

**Pull request merged**